ansible_runner/runner.py

```diff
@@ -286,6 +286,20 @@ def _delete(retries=15):
                         raise
                 return True
             _delete()
+
+        # Record installed Collections + Collections version info
+        cmd = 'ansible-galaxy collection list -o json'
+        proc = Popen(cmd, stdout=PIPE, stderr=PIPE, shell=True)
+        stdout, stderr = proc.communicate()
+        if proc.returncode:
+            # Unable to create cgroup
+            logger.error('Unable to list Collections: {}'.format(stderr))
+            raise RuntimeError('Unable to list Collections: {}'.format(stderr))
+        else:
+            logger.info("Listing Collections '{}'".format(stdout))
+        with open(os.path.join(self.config.artifact_dir, "collections.json"), "wb") as f:
```

Comment:
Instead of running popen and managing the data via a pipe, you could just have Popen write directly to your file handle, and skip some of this:

```python
>>> subprocess.Popen(['ls', '-la'], stdout=open('testing123', 'wb')).communicate()
(None, None)
>>>
~ cat testing123
```

Reply:
We want to keep this as-is so we can detect if there's a return code, and only create the `collections.json` file if the command was successful.